plugins/modules/sonic_ntp.py

```diff
@@ -0,0 +1,327 @@
+
+# After state:
+# ------------
+#
+#sonic# show ntp server
+#----------------------------------------------------------------------
+#NTP Servers                     minpoll maxpoll Authentication key ID
+#----------------------------------------------------------------------
+#10.11.0.1                       6       10
+#dell.com                        6       9
+#
+#
+# Using deleted
+#
+# Before state:
+# -------------
+#
+#sonic# show ntp global
+#----------------------------------------------
+#NTP Global Configuration
+#----------------------------------------------
+#NTP source-interfaces:  Ethernet0, Ethernet4, Ethernet8, Ethernet16
+#
+- name: Delete NTP source-interface configuration
+  ntp:
+    config:
+      source_interfaces:
+        - Ethernet8
+        - Ethernet16
+    state: deleted
+
+# After state:
+# ------------
+#
+#sonic# show ntp global
+#----------------------------------------------
+#NTP Global Configuration
+#----------------------------------------------
+#NTP source-interfaces:  Ethernet0, Ethernet4
+#
+#
+# Using deleted
+#
+#
+- name: Delete NTP key configuration
+  ntp:
+    config:
+      ntp_keys:
+        - key_id: 10
+        - key_id: 20
+    state: deleted
+
+# Using merged
+#
+# Before state:
+# -------------
+#
+#sonic# show ntp server
+#----------------------------------------------------------------------
+#NTP Servers                     minpoll maxpoll Authentication key ID
+#----------------------------------------------------------------------
+#10.11.0.1                       6       10
+#dell.com                        6       9
+#
+- name: Merge NTP server configuration
+  ntp:
+    config:
+      servers:
+        - address: 10.11.0.2
+          minpoll: 5
+        - address: dell.org
+          minpoll: 7
+          maxpoll: 10
+    state: merged
+
+# After state:
+# ------------
+#
+#sonic# show ntp server
+#----------------------------------------------------------------------
+#NTP Servers                     minpoll maxpoll Authentication key ID
+#----------------------------------------------------------------------
+#10.11.0.1                       6       10
+#10.11.0.2                       5       9
+#dell.com                        6       9
+#dell.org                        7       10
+#
+#
+# Using merged
+#
+# Before state:
+# -------------
+#
+#sonic# show ntp global
+#----------------------------------------------
+#NTP Global Configuration
+#----------------------------------------------
+#NTP source-interfaces:  Ethernet0, Ethernet4
+#
+- name: Merge NTP source-interface configuration
+  ntp:
+    config:
+      source_interfaces:
+        - Ethernet8
+        - Ethernet16
+    state: merged
+
+# After state:
+# ------------
+#
+#sonic# show ntp global
+#----------------------------------------------
+#NTP Global Configuration
+#----------------------------------------------
+#NTP source-interfaces:  Ethernet0, Ethernet4, Ethernet8, Ethernet16
+#
+#
+# Using merged
+#
+#
```

Comment:
Please add displaying of before/after state for key configuration for both this "merged" exampel and also for the "deleted" example (above). 

The NTP- specific "show" commands used above for the server and source interface before/after states are good. But, if there is no NTP-specific "show" command that will display the key configuration, please just use "show running-configuration" for the before/after state displays.

Reply:
Added "before" and "after" states.

Reply:
Thank you.